### PR TITLE
Leangseu edx/masquerade user

### PR DESCRIPTION
**Pull request by leangseu-edx**
https://user-images.githubusercontent.com/83240113/191574872-e0a22fcf-90b8-4b01-b2d4-4b0da805b639.mov

Screen Shot 2022-09-20 at 3 07 24 PM
Screen Shot 2022-09-20 at 3 07 49 PM
Screen Shot 2022-09-20 at 3 07 09 PM
Screen Shot 2022-09-20 at 3 06 50 PM

### CODE REVIEW

File: src/containers/MasqueradeBar/index.scss [hunk 1]
```diff
@@ -3,11 +3,17 @@
 .masquerade-bar {
   display: flex;
   align-items: flex-start;
-  color: #fff;
   padding: map-get($spacers, 2) map-get($spacers, 3);
 
   .masquerade-form-label {
     padding: map-get($spacers, 2) map-get($spacers, 1);
+    color: #fff;
```
Comment (muselesscreator) [inline src/containers/MasqueradeBar/index.scss:10]: Should be loading colors from theme, not defining manually if possible.

File: src/data/services/lms/api.js [hunk 2]
```diff
@@ -21,8 +21,19 @@ const deleteEntitlementEnrollment = ({ uuid }) => client().delete(stringifyUrl(
   { course_run_id: null },
 ));
 
+const masqueradeAs = ({ user }) => new Promise((resolve, reject) => {
+  if (!user) {
+    reject('username or email is required'); // eslint-disable-line prefer-promise-reject-errors
```
Comment (muselesscreator) [inline src/data/services/lms/api.js:26]: I would put this as an option in the init api call.

File: src/containers/MasqueradeBar/hooks.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export const useMasqueradeInput = () => {
+  const [masqueradeInput, setMasqueradeInput] = module.state.masqueradeInput('');
+  const handleMasqueradeInputChange = (e) => setMasqueradeInput(e.target.value);
+  return {
+    handleMasqueradeInputChange,
+    masqueradeInput,
+  };
+};
+
+export const useMasqueradeBar = ({
```
Comment (muselesscreator) [inline src/containers/MasqueradeBar/hooks.js:22]: nit: `useMasqueradeBarData`.  Prefer when the hooks describe what they are returning.  this phrasing sounds like it should return a component.

File: src/containers/MasqueradeBar/hooks.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  authenticatedUser,
+}) => {
+  const dispatch = useDispatch();
+  const { formatMessage } = useIntl();
+  const canMasquerade = authenticatedUser?.roles.includes('staff') || authenticatedUser?.administrator;
```
Comment (muselesscreator) [inline src/containers/MasqueradeBar/hooks.js:27]: I don't think all staff are/should-be able to masquerade

Comment (leangseu-edx) [reply src/containers/MasqueradeBar/hooks.js:27]: I did this based on the backend's api at the moment.

**Review by muselesscreator — APPROVED** [review]

**Pull request merged**